Core/GDCore/Project/EventsFunction.h

```diff
@@ -116,6 +117,24 @@ class GD_CORE_API EventsFunction {
    */
   FunctionType GetFunctionType() const { return functionType; };
 
+  /**
+   * \brief Returns true if the function is private.
+   */
+  bool IsPrivate() {
+    if (isPrivate)
+      return true;
+    else
+      return false;
+  }
+
+  /**
+   * \brief Toggles the privateness of the function.
+   */
+  EventsFunction& TogglePrivate() {
```

Comment:
Nitpicking: In general avoid designing classes with `ToggleSomeState`: prefer the more explicit `SetSomeState`.
The reason is because most of the time your client/user will want to set something to a specific value. For example, if I click on a button to make my function private, my goal is not to "toggle the privateness", but to set it to true. If I want to make something public, I don't care about toggling anything, I want to set it to false (and if it was false before, well I still want it false, not true :)).

It's also true in the case of UI framework where some update could be batched: the update should contain the definitive value you want to apply, not a command to toggle - which could be the contrary of what you wish if for some reason the value was inverted in the meantime.